src/block/compress.rs

```diff
@@ -252,69 +242,61 @@ fn count_same_bytes(input: &[u8], candidate: usize, cur: &mut usize) -> usize {
 
 /// Write an integer to the output in LSIC format.
 #[inline]
-#[cfg(feature = "safe-encode")]
 fn write_integer(output: &mut Sink, mut n: usize) {
-    while n >= 0xFF {
-        n -= 0xFF;
-        push_byte(output, 0xFF);
+    // A predictable branch for the common case
+    if n < 0xFF {
+        push_byte(output, n as u8);
+        return;
     }
 
-    // Write the remaining byte.
-    push_byte(output, n as u8);
-}
-
-/// Write an integer to the output in LSIC format.
-#[inline]
-#[cfg(not(feature = "safe-encode"))]
-fn write_integer(output: &mut Sink, mut n: usize) {
-    // Write the 0xFF bytes as long as the integer is higher than said value.
-    push_u32(output, 0xFFFFFFFF);
+    // Write the 0xFF bytes in chunks of 4.
     while n >= 4 * 0xFF {
         n -= 4 * 0xFF;
-        output.set_pos(output.pos() + 4);
         push_u32(output, 0xFFFFFFFF);
     }
 
-    // Updating output len for the remainder
-    output.set_pos(output.pos() + 1 + n / 255);
-    unsafe {
-        // Write the remaining byte.
-        *output.as_mut_ptr().sub(1) = (n % 255) as u8;
-    };
-}
-
-#[inline]
-#[cfg(not(feature = "safe-encode"))]
-fn push_u32(output: &mut Sink, el: u32) {
-    unsafe {
-        let out_ptr = output.as_mut_ptr();
-        core::ptr::copy_nonoverlapping(el.to_le_bytes().as_ptr(), out_ptr, 4);
-    }
+    // Write the last 0..=3 0xFF bytes + remainder.
+    let ffs = n / 0xFF;
```

Comment:
what does ffs mean?

Reply:
It's meant to be the number of `0xFF`s to be output. Sorry :facepalm: quite bad naming from my part.